dtreeviz/models/shadow_decision_tree.py

```diff
@@ -438,17 +444,26 @@ def _get_y_data(y_data):
 
     @staticmethod
     def get_shadow_tree(tree_model, x_data, y_data, feature_names, target_name, class_names=None, tree_index=None):
+        if hasattr(tree_model, 'get_booster'):
```

Comment:
I'm not sure what this is doing?  Are we trying to do two things at once? merge XGBoost and Make them optional? Why wasn't this in there before is my question if we already support XGBoost

Reply:
This basically allows users to pass the sklearn compatible wrappers `xgboost.sklearn.XGBClassifier` and `xgboost.sklearn.XGBRegressor`. Both have a `get_booster()` method to get the underlying xgboost model. 

Otherwise users would just get weird error messages and have to figure out they have to do `.get_booster()` themselves.

Reply:
indeed, it was a little confuse to me also... but it's the code to support https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier, which is a sklearn API for xgboost.  
@oegedijk can you confirm this ?

If this line of code was for you also confusing, we should refactor it somehow

Reply:
a more explicit way of doing it would be `if str(type(model)).endswith("XGBClassifier'>"): model=model.get_booster()`

Reply:
yes maybe a more explicit version is better. I think what really is lacking is a simple comment indicating what that code does ;)

Reply:
Also instead of `str(type(...))` I wonder if it's more readable or explicit anyway to use `model.__class__.__name__`??